doc/man3/X509_STORE_CTX_new.pod

```diff
@@ -229,6 +237,14 @@ at the same time. During verification, this trust setting will be verified
 to check whether it is consistent with the trust set by the system administrator
 for certificates in the chain.
 
+The purpose however is only set by X509_STORE_CTX_set_purpose(), if no purpose
+was set previously. If another purpose was already set, the new purpose is
+not set but silently ignored.
+
+X509_STORE_CTX_set_purpose_ex() extends X509_STORE_CTX_set_purpose() by
+explicitly overriding any existing purpose setting
+(and any trust implied by the purpose).
```

Comment:
Why did you write this part?
The function treats trust and purpose independently.